### PR TITLE
Added multiple functions

### PR DESCRIPTION
**Pull request by Thefoxy1978**

### CODE REVIEW

**Review by nedex — APPROVED** [review]

**Review by nedex — APPROVED** [review]

**Review by nedex — APPROVED** [review]

File: src/game/91F8C.c [hunk 1]
```diff
@@ -50,7 +66,38 @@ INCLUDE_ASM("asm/game/nonmatchings/91F8C", func_800B5858);
...
+                var_a1 += 1;
+                var_a2 -= 1;
+                *var_a0 = temp_v0;
+                var_a0 += 1;
+            } while (var_a2 > 0);
+        }
+        var_v0 = temp_v1;
+    }
+    return var_v0;
+}
```
Comment (sonicdcer) [inline src/game/91F8C.c:99]:
```suggestion
u8* func_800B5878(u8* arg0, u8* arg1, s32 arg2) {
    u8* ret = NULL;
    u8* temp_v1;
    
    if (arg0 != NULL) {
        temp_v1 = arg0;
        while (arg2 > 0) {
            *arg0 = *arg1;
            arg1++;
            arg2--;
            arg0++;
        }
        ret = temp_v1;
    }
    return ret;
}
```
simplified this function a bit

File: src/game/99ADC.c [hunk 2]
```diff
@@ -36,7 +36,14 @@ INCLUDE_ASM("asm/game/nonmatchings/99ADC", func_800BB500);
 
 INCLUDE_ASM("asm/game/nonmatchings/99ADC", func_800BB5D8);
 
+// compiles with 2.7.2
```
Comment (sonicdcer) [inline src/game/99ADC.c:39]:
```suggestion
// matches with gcc 2.7.2

File: src/game/99ADC.c [hunk 3]
```diff
@@ -92,8 +99,22 @@ INCLUDE_ASM("asm/game/nonmatchings/99ADC", func_800BC41C);
...
+#ifndef NON_MATCHING
 INCLUDE_ASM("asm/game/nonmatchings/99ADC", func_800BC518);
-
+#else
+s32 func_800BC518(void) {
+    *D_800EC9D8 = (*D_800EC9D8 & 0xF0FFFFFF) | 0x20000000;
+}
+#endif
+
+// compiles with v0 instead of a0
```
Comment (sonicdcer) [inline src/game/99ADC.c:111]:
```suggestion
// matches with gcc 2.7.2

**Review by sonicdcer — APPROVED** [review]
Congratulations on your first PR on Medievil-Decomp and thank you for helping us reverse engineer this game!

I left a few nits to be addressed :)

**Pull request merged**